src/dune_engine/rules.mli

```diff
@@ -29,7 +29,10 @@ module Dir_rules : sig
         Action of
           Rule.Anonymous_action.t Action_builder.t
 
-    type t = { expansions : (Loc.t * item) Appendable_list.t } [@@unboxed]
+    type t
```

Comment:
@snowleopard I think we should keep this refactoring. It's not strictly necessary, but it's important to remember that we need to get the list of rules in reverse order (using `to_list_rev`) to avoid an unnecessary allocation. I think that detail should stay private to `Rules`.

Reply:
Changing `to_list` to `to_list_rev` seems fine but the current change is quite a bit bigger.

Reply:
Okay, reverted as well.

Reply:
Thanks!